process/update.go

```diff
@@ -171,40 +196,33 @@ func (pm *Manager) updateConfig(ctx context.Context, newConfig *config.Config) e
 // stopChangedProcesses stops any processes that need to be updated.
 // It returns a map keyed by the names of all such processes.
 func (pm *Manager) stopChangedProcesses(ctx context.Context, newConfig *config.Config) (map[string]bool, error) {
-	changed := make(map[string]bool)
-	for name, oldp := range pm.config.Programs {
-		newp := newConfig.Programs[name]
-		if reflect.DeepEqual(newp, oldp) {
-			// TODO we could potentially be less conservative and avoid
-			// tearing down the process when a new command doesn't actually
-			// need to be run.
-			continue
+	return pm.stopProcesses(ctx, func(p *config.Program) bool {
+		return !reflect.DeepEqual(newConfig.Programs[p.Name], pm.config.Programs[p.Name])
```

Comment:
I imagine a future version might perform a more explicit comparison to ignore some changes that don't require a restart, like labels, autostart, etc?

Reply:
Yes, definitely.